brigade/plugins/tasks/networking/napalm_configure.py

```diff
@@ -25,13 +25,15 @@ def napalm_configure(task, configuration, replace=False, hostname=None, username
           * diff (``string``): change in the system
     """
     parameters = {
-        "hostname": hostname or task.host["brigade_ip"],
-        "username": username or task.host["brigade_username"],
-        "password": password or task.host["brigade_password"],
+        "hostname": hostname or task.host,
+        "username": username or task.host.username,
+        "password": password or task.host.password,
         "timeout": timeout,
-        "optional_args": optional_args or {"port": task.host["napalm_port"]},
+        "optional_args": optional_args or {},
     }
-    network_driver = get_network_driver(driver or task.host["nos"])
+    if "port" not in parameters["optional_args"] and task.host.network_api_port:
+        parameters["optional_args"] = task.host.network_api_port
```

Comment:
This should be `parameters["optional_args"]["port"] = task.host.network_api_port` for all napalm tasks. However it might also be confusing for some people when there's also an `ssh_port` property on the host. `network_api_port` is more correct in this case though.